keda/assets/logs/keda.yaml

```
@@ -0,0 +1,61 @@
+id: keda
+metric_id: keda
+backend_only: false
+installation_sources:
+  - keda
+  - keda-admission-webhooks
+  - keda-metrics-apiserver
+facets:
+pipeline:
+  type: pipeline
+  name: Keda
+  enabled: true
+  filter:
+    query: source:(keda OR keda-admission-webhooks OR keda-metrics-apiserver)
+  processors:
+    - type: grok-parser
+      name:  Keda Parser
+      enabled: true
+      source: message
+      samples:
+        - 'I1216 14:53:06.261024       1 main.go:112] "msg"="Connecting Metrics Service gRPC client to the server" "address"="keda-operator.keda.svc.cluster.local:9666" "logger"="keda_metrics_adapter"'
+        - 'I1216 14:53:06.680302       1 tlsconfig.go:243] "Starting DynamicServingCertificateController"'
+        - '2024-12-23T14:21:51Z	ERROR	scale_handler	error resolving auth params	{"type": "ScaledJob", "namespace": "default", "name": "csv-processor", "triggerIndex": 0, "error": "connection to redis failed: dial tcp 10.244.0.6:6379: connect: no route to host"}'
+        - '2024-12-23T14:21:38Z	INFO	KubeAPIWarningLogger	unknown field "status.authenticationsTypes"'
+        - '2024-12-16T14:53:12Z	INFO	controller-runtime.builder	Registering a validating webhook	{"GVK": "keda.sh/v1alpha1, Kind=ScaledObject", "path": "/validate-keda-sh-v1alpha1-scaledobject"}'
+      grok:
+        matchRules: |
+          keda_metrics_adapter3 %{_level_adapter}%{_date_adapter}\s+%{_pid}.*\s+%{_logger_adapter}]\s+"msg"="%{_msg}" %{data:component:keyvalue}
```

```suggestion
          keda_metrics_adapter3 %{_level_adapter}%{_date_adapter}\s+%{_pid}\s+%{_logger_adapter}]\s+"msg"=%{doubleQuotedString:msg} %{data:component:keyvalue}
```

- Not sure if there is a reason for `.*`? 
- I think we could also make `_msg` be a `doubleQuotedString` (better than `data`). You won't be able to share the same supportRule with the keda_operator rules but that also makes sense since msg does not have the same format.

---

I think a lot of the `.*` is because I couldn't get it to work otherwise I can revisit this with your suggestion and see.

---

The doublequoted string here seemed to not be working for me. There was one log that I was targetting with this rule for example:

```
I1223 01:15:24.279269       1 leaderelection.go:254] attempting to acquire leader lease keda/operator.keda.sh...
```

Which as you can see it isn't a doublequoted string here.

---

Just tried to remove all of those and parsing still works fine on my end so I'd remove them unless you have extra samples that do not pass without those.